torch/csrc/jit/passes/quantization.cpp

```diff
@@ -1917,6 +1978,57 @@ void FoldConvBatchNorm2dHelper::transform() {
   }
 }
 
+void replaceConv2dBiasWithGetAttr(Module& module) {
+  auto graph = module.get_method("forward").graph();
+  // Only looks fors _convolution pattern.
+  // Thus assumes that tracing will have always gotten rid of aten::conv2d.
+  // If it did not, BN folding will fail.
+  const PatternInfo& pattern_convolution = PatternInfo::parse_from_str(R"(
+      graph(%a, %w, %b, %stride:int[], %padding:int[], %dilation:int[],
+          %transposed:bool, %output_padding:int[], %groups:int, %benchmark:bool,
+          %deterministic:bool, %cudnn_enabled:bool):
+        %conv_out = aten::_convolution(%a, %w, %b, %stride, %padding, %dilation,
+            %transposed, %output_padding, %groups, %benchmark, %deterministic, %cudnn_enabled)
+        return (%conv_out) )");
+  const Graph& pattern_convolution_graph = *pattern_convolution.pattern_graph;
+  const auto& convolution_vmap = pattern_convolution.vmap;
+
+  const auto& matches = findPatternMatches(pattern_convolution_graph, *graph);
+  for (const auto& match : matches) {
+    // We come here only if the bias was not present in the module.
+    // In that case, the corresponding graph will not have getAttr("bias")
+    // Insert that in the graph.
+    // And change _convolution to take the new value.
+    auto conv_node = match.values_map.at(convolution_vmap.at("conv_out"))->node();
+    WithInsertPoint ins(conv_node);
+    Value* bias_attr_val =
+      graph->insertGetAttr(graph->inputs()[0], "bias")
+          ->setType(TensorType::get());
+    constexpr size_t conv_bias_index = 2;
+    conv_node->replaceInput(conv_bias_index, bias_attr_val);
+  }
+}
+
+void addBiasForConv2dIfNone(Module& module) {
```

Comment:
also I think it's probably more reliable to first find the _convolution pattern in the forward and also check for the type name `Conv2d`, because people might write modules end with name `Conv2d` and do something else in the forward function

Reply:
Well if that is the case, lot of thing will break down. Now you will have to check that _convolution is the only thing in there AND that it is for Conv2d. But then you may also miss cases like 
```
class XYZConv2d(nn.Module):
.....
   def forward(input):
      y = dosomethingfunky(input)
      z = _convolution(y)
      return z
```
Assuming we dont care of cases like the one above, we now have to ensure there is nothing else being called inside module besides _convolution. I dont think this is scalable. I think instead of pattern matching on "Conv2d" we should just do ".Conv2d". The difference of "." should filter out custom modules with Conv2d in their name. Ideally it should be "nn.modules.Conv2d"? But I dont think that works for tracing.

Honestly if we really want to cover all cases of aten::conv2d and aten::batch_norm fusion, we should do it like the other PR with graph inlining and freezing. That ensures we catch all cases despite people writing funky modules like above.

Reply:
Yeah, I'm just saying it will be more reliable, not saying that it will be able to handle all cases. But I think adding more strict filtering for the name should help as well, why `.Conv2d` doesn't work for tracing?

Reply:
".Conv2d" would work, just that I would have liked to be more explicit with `nn.module.Conv2d` and that wont work. For now I will go with ".Conv2d" and add test case to make sure we dont do fusion on other types that contain Conv2d.

Reply:
please add a comment that this is to make sure our matching is more reliable to user defined modules that end with Conv2d.